Lib/test/test_optimizer.py

```diff
@@ -78,6 +79,40 @@ def func(x=0):
             )
 
 
+class TestOptimizations(unittest.TestCase):
+
+    def test_globals_to_consts(self):
```

Comment:
Note that this test only fails when tier 2 is selected (e.g. `-X uops`, or when the JIT is enabled).

We have machinery in test_capi/test_opt.py to force the optimizer on. It feels pretty arbitrary whether this test belongs here or there. Can you explain the criteria for where a tier-2-related test should go?

Reply:
> Can you explain the criteria for where a tier-2-related test should go?

I don't think there are any. This test should pass for tier-1 and tier-2, as should all tests.

Reply:
Okay. In my mind this test should be moved to `test_capi/test_opt.py` and should be wrapped in `with temporary_optimizer(_testinternalcapi.new_uop_optimizer()):`. Otherwise it doesn't test anything unless build with the JIT or run with `-Xuops`.